smurfs/src/components/InputForm.js

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { connect }  from "react-redux";
+import { addSmurf } from "../actions"
+
+class InputForm extends React.Component {
+    constructor(){
+        super();
+        this.state = {
+            name   : " ",
+            age    : " ",
+            height : " "
+        }
+    }
+    changeHandler = e => {
+        this.setState({
+          ...this.state,
+          [e.target.name]: e.target.value
+        });
+      }
+    submitHandler = e => {
+        e.preventDefault();
+        const { name, age, height } = this.state;
+        this.props.addSmurf({ name, age, height });
```

Comment:
Because it's only the 3 things you are sending on state, you can pass the entire state and get the same result, this.props.addSmurf(this.state)